shoop/front/models/stored_basket.py

```diff
@@ -0,0 +1,41 @@
+# -*- coding: utf-8 -*-
+# This file is part of Shoop.
+#
+# Copyright (c) 2012-2015, Shoop Ltd. All rights reserved.
+#
+# This source code is licensed under the AGPLv3 license found in the
+# LICENSE file in the root directory of this source tree.
+
+from django.conf import settings
+from django.db import models
+from shoop.core.fields import MoneyField, TaggedJSONField
+import random
+
+ID_KEYSPACE = "0123456789abcdefghijklmnopqrstuvwxyz"
+
+
+def generate_key():
+    return "".join(random.choice(ID_KEYSPACE) for x in range(32))
+
+
+class StoredBasket(models.Model):
+    # A combination of the PK and key is used to retrieve a basket for session situations.
+    key = models.CharField(max_length=32, default=generate_key)
```

Comment:
It seems that my concerns about predetermination of basket keys aren't so far fetched after all. Here's a quote from http://www.quadibloc.com/crypto/co4814.htm

> Despite the fact that the Mersenne Twister is an extremely good pseudo-random number generator, it is not cryptographically secure by itself for a very simple reason. It is possible to determine all future states of the generator from the state the generator has at any given time, and either 624 32-bit outputs, or 19,937 one-bit outputs are sufficient to provide that state. Using a cryptographically-secure hash function, such as SHA-1, on the output of the Mersenne Twister has been recommended as one way of obtaining a keystream useful in cryptography.